src/app/utils/state.ts

```diff
@@ -92,10 +92,16 @@ class StateManager {
     const { text } = result!;
     const resultJsonMetadata = JSON.parse(text);
     // TODO - validate or retry here
+    resultJsonMetadata.poop = resultJsonMetadata.poop
+      ? resultJsonMetadata.poop
+      : 0;
 
     await this.updateTamagotchiStatus({
       ...resultJsonMetadata,
-      poop: status.poop + resultJsonMetadata.poop,
+      poop:
+        parseInt(status.poop) > parseInt(resultJsonMetadata.poop)
```

Comment:
**issue:** Add error handling for parseInt operations

Consider using Number() or adding fallback values in case parsing fails